src/codeu/chat/client/ClientUser.java

```diff
@@ -51,8 +55,9 @@ static public boolean isValidName(String userName) {
       clean = false;
     } else {
 
-      // TODO: check for invalid characters
-
+      Pattern pattern = Pattern.compile("^[a-zA-Z0-9]*$");
```

Comment:
Make this compiled pattern a static constant on this class, so that it only has to be compiled once.